apps/mobile-wallet/src/components/NFTsGrid.tsx

```diff
@@ -35,41 +36,57 @@ interface NFTsGridProps extends ModalContentProps {
   nftSize?: number
 }
 
-const gap = 12
-const screenPadding = 20
+const gap = DEFAULT_MARGIN / 2
+const containerHorizontalPadding = DEFAULT_MARGIN - gap
 
-const NFTsGrid = ({ addressHash, nfts: nftsProp, nftSize, nftsPerRow = 3, ...props }: NFTsGridProps) => {
+const NFTsGrid = ({ addressHash, nfts: nftsProp, nftSize, nftsPerRow = 3 }: NFTsGridProps) => {
   const selectAddressesNFTs = useMemo(makeSelectAddressesNFTs, [])
   const nfts = useAppSelector((s) => selectAddressesNFTs(s, addressHash))
+  const isLoadingNfts = useAppSelector((s) => s.nfts.loading)
   const theme = useTheme()
-  const { t } = useTranslation()
 
   const data = nftsProp ?? nfts
-  const columns = data.length < nftsPerRow ? data.length : nftsPerRow
+  const columns = nftsPerRow
   const { width: windowWidth } = Dimensions.get('window')
-  const totalGapSize = (columns - 1) * gap + screenPadding * 2
+  const totalGapSize = columns * gap * 2 + containerHorizontalPadding * 2
   const size = nftSize ?? (windowWidth - totalGapSize) / columns
 
   return (
-    <ModalFlatListContent
+    <FlashList
       data={data}
-      verticalGap
       keyExtractor={(item) => item.id}
-      renderItem={({ item: nft }) => <NFTThumbnail key={nft.id} nftId={nft.id} size={size} />}
+      renderItem={({ item: nft }) => (
+        <NFTThumbnailContainer>
```

Comment:
Shouldn't the `key` be here instead?

Reply:
Right, good catch, thanks.

Reply:
Fixed.

Reply:
<img width="515" alt="image" src="https://github.com/user-attachments/assets/60f51cf0-f1e8-4467-8bb2-13738d2678fc">

It doesn't look fixed